Allow environment overrides specifically for Storybook

DingoEatingFuzz opened the pull request:
Stories are rendered as full Ember Apps including the default environment, which may have settings that aren't appropriate for only rendering a single story in an iframe.

This makes it so settings in `environment.js` can be overridden by settings in `.storybook/environment.js`.

It also resets `rootURL` to `/` (after doing the existing `rootURL` resource transformations) since Ember doesn't at all like running the router in an iframe when `rootURL` is anything other than `/`.

gabrielcsapo commented on index.js:
```diff
@@ -93,6 +94,22 @@ module.exports = {
 
     this.ui.writeDebugLine('Generating preview-head.html');
 
+    const environment = parsedConfig.meta.find(meta => meta.name.endsWith('config/environment'));
```
Can we write a test case for this?

DingoEatingFuzz replied:
Yep! Just wanted to make sure you were cool with the feature first 😅

DingoEatingFuzz commented:
Sorry about the delay. I rebased and added some tests.

There weren't existing tests for index.js (which I get, since it would require mocking all the build stuff) so I refactored the implementation a bit to push more of the logic into utils. Hopefully that is sufficient coverage.

gossi commented:
Looking good to me. Going to hit the merge button 👍
